software/controller/lib/hal/hal_stm32.cpp

```diff
@@ -83,40 +57,27 @@ extern "C" void abort() {
   };
 }
 
-// This function is called _init() above.  It does some basic
-// chip initialization.
+// This function is called by _init() above.  It does some basic chip initialization.
 //
-// The main things done here are to enable the FPU because if
-// we don't do that then we'll get a fatal exception if any
-// constructor uses any floating point math, and to enable
-// the PLL so we can run at full speed (80MHz) rather then the
-// default speed of 4MHz.
+// The main things done here are to enable the FPU because if we don't do that then we'll get a
+// fatal exception if any constructor uses any floating point math, and to enable the PLL, so we can
+// run at full speed (80MHz) rather than the default speed of 4MHz.
 void HalApi::EarlyInit() {
-  // Enable the FPU.  This allows floating point to be used without
-  // generating a hard fault.
-  // The system control registers are documented in [PM] chapter 4.
-  // Details on enabling the FPU are in section 4.6.6.
-  SysControlReg *sys_ctl = SysControlBase;
-  sys_ctl->coproc_access_control = 0x00F00000;
-
-  /// \TODO initialized but never used. Do we still need it?
+  SysControl::enable_fpu();
```

Comment:
Maybe keep the rationale (first two lines of that comment) as to why we enable the FPU here, even if the part about registers move to where they are abstracted away?

Reply:
ok